A/ADOLC/build_tarballs.jl

```diff
@@ -0,0 +1,97 @@
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+# See https://discourse.julialang.org/t/binarybuilder-jl-cant-dlopen-because-of-libopenlibm-so/108486
+# This build script needs to be run with julia 1.7
+
+# See https://github.com/JuliaLang/Pkg.jl/issues/2942
+# Once this Pkg issue is resolved, this must be removed
+uuid = Base.UUID("a83860b7-747b-57cf-bf1f-3e79990d037f")
+delete!(Pkg.Types.get_last_stdlibs(v"1.6.3"), uuid)
+
+name = "ADOLC"
+version = v"0.0.24"
+
+isyggdrasil = get(ENV, "YGGDRASIL", "") == "true"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/coin-or/ADOL-C.git", "b7ea10536a1e452f7dd47a90ef8c8118ce3e8432"),
+    GitSource("https://github.com/TimSiebert1/libadolccxx.git", "51c23abf9ee2e1566331b440e81503ac9e7e8f5b"),
+]
+
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+cd ADOL-C/ && autoreconf -fi && ./configure --enable-atrig-erf --prefix=${prefix} --build=${MACHTYPE} --host=${target}
+make -j${nproc}
+make install
+mkdir ${prefix}/share/licenses/ADOLC
+install_license LICENSE
+
+cd ../libadolccxx/
+cmake -DCMAKE_INSTALL_PREFIX=${prefix}\
+      -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN}\
+      -DCMAKE_BUILD_TYPE=Release\
+      -DADOLC_DIR=${WORKSPACE}/destdir\
+      -DJulia_PREFIX=${prefix}\
+      -B build .
+
+cd build
+make -j${nprocs}
+make install
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+
+platforms = [
+    Platform("x86_64", "linux"; libc = "glibc"),
+    Platform("aarch64", "macos"; ),
+    Platform("x86_64", "windows"; )
+    # Platform("i686", "linux"; libc = "glibc"),
+    # Platform("aarch64", "linux"; libc = "glibc"),
+    # Platform("armv6l", "linux"; call_abi = "eabihf", libc = "glibc"),
+    # Platform("armv7l", "linux"; call_abi = "eabihf", libc = "glibc"),
+    # Platform("powerpc64le", "linux"; libc = "glibc"),
+    # Platform("x86_64", "linux"; libc = "musl"),
+    # Platform("x86_64", "macos"; ),
+    # Platform("x86_64", "freebsd"; ),
+    # Platform("i686", "windows"; ),
+]
+
+# See https://discourse.julialang.org/t/binarybuilder-jl-cant-dlopen-because-of-libopenlibm-so/108486
+# It seems we need a separate build for each Julia version
+if isyggdrasil
+    include("../../L/libjulia/common.jl")
+else
+    # For local test, download
+    # https://github.com/JuliaPackaging/Yggdrasil/blob/master/L/libjulia/common.jl
+    # https://github.com/JuliaPackaging/Yggdrasil/blob/master/fancy_toys.jl
+    include("common.jl")
+end
+julia_versions=VersionNumber[v"1.9.0", v"1.10.0"]
+platforms = vcat(libjulia_platforms.(julia_versions)...)
+
+# Platform for initial testing
+platforms = filter( p-> (arch(p)=="x86_64" && os(p)=="linux" && libc(p)=="glibc")
+                    || os(p)=="macos"
+                    || os(p)=="windows",
+                    platforms)
```

Comment:
Since you hard coded a small list of platforms above, this isn't really doing anything. But as stated, I would recommend to instead start with `platforms = supported_platforms()` and then gradually remove things via `filter!`. Alternatively, use `supported_platforms(; exclude=...)`. There are plenty examples for each in Yggdrasil, just search a bit around.

(If you really need to keep this, I would at least advise against using `os(p)=="macos"` -- instead use `Sys.isapple(p)`. Similarly you can use `Sys.iswindows(p)`.)